src/maybe.ts

```diff
@@ -15,6 +15,17 @@ export enum Variant {
   Nothing = 'Nothing',
 }
 
+export interface JustJsonShape {
+  variant: Variant.Just;
+  value: any;
```

Comment:
Following that, this would be `value: T`:

```suggestion
  value: T;
```